kartothek/io_components/utils.py

```diff
@@ -19,6 +19,8 @@
 except ImportError:
     from typing import Literal  # type: ignore
 
+# Literal false is sentinel, see function body of `_ensure_compatible_indices` for details
+INFERRED_INDICES = Union[Literal[False], List[str]]
```

Comment:
I think this should be camelcase, to be consistent with how python usually writes types: `InferredIndices`